include/ck/utility/numeric_limits.hpp

```diff
@@ -0,0 +1,555 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2025, Advanced Micro Devices, Inc. All rights reserved.
+#pragma once
+#include "ck/utility/data_type.hpp"
+
+namespace ck {
+
+#if defined(__HIPCC_RTC__) || defined(CK_CODE_GEN_RTC)
+template <typename T>
+struct NumericLimits;
+
+template <>
+struct NumericLimits<int32_t>
```

Comment:
should `NumericLimits<T>` be defined in `T_utils.hpp` for each datatype?

Reply:
I guess our ultimate goal is to split all data type declarations in separate headers like https://github.com/ROCm/composable_kernel/blob/develop/include/ck/utility/e8m0.hpp. In such case, would it make sense to have numeric_utils and numeric_limits declarations in numeric_utils.hpp and numeric_limits.hpp, and add their specializations in corresponding data type headers?

Reply:
Well, it looks like the numeric limits and numeric utils headers are needed just in a handful of places, so even when we further split up the data_type header by types, we can still keep those headers separate.

Reply:
Sounds fine to me, but we'll have to include all data type declaration headers in numeric_utils.hpp and numeric_limits.hpp.